src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -271,11 +272,43 @@ public AzureAppConfigurationOptions Connect(Uri endpoint, TokenCredential creden
             }
 
             ConnectionString = null;
-            Endpoint = endpoint;
+            Endpoints = new List<Uri>() { endpoint };
             Credential = credential;
             return this;
         }
 
+        /// <summary>
+        /// Connect the provider to Azure App Configuration and their replicas using list of endpoints and token credential.
+        /// </summary>
+        /// <param name="endpoints">The list of endpoints of the Azure App Configuration and it's replicas to connect to.</param>
+        /// <param name="credential">Token credential to use to connect.</param>
+        public AzureAppConfigurationOptions Connect(IEnumerable<Uri> endpoints, TokenCredential credential)
+        {
+            if (endpoints == null || endpoints.Count() == 0)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            IEnumerable<string> configStoreNames = endpoints.Select(endpoint => endpoint.Host.Split('.')[0]);
+            var primaryStores = configStoreNames.Where(storeName => !storeName.Contains(ConfigurationStoreConstants.ConfigStoreNameReplicaSeparator));
+
+            if (primaryStores.Count() > 1)
+            {
+                throw new ArgumentException($"{nameof(endpoints)} can have only one primary configuration store endpoint.");
+            }
+
+            if (!configStoreNames.All(storeName => storeName.Split(configStoreNameSplitSeparator, StringSplitOptions.None)[0].Equals(primaryStores.Single(), StringComparison.OrdinalIgnoreCase)))
+            {
+                throw new ArgumentException($"{nameof(endpoints)} can only contain endpoints for the primary configuration store and it's replicas.");
+            }
```

Comment:
We have only one rule that is all endpoints must belong to the same App Configuration store. I hope we can combine the exception throwing in one place.

BTW, we shouldn't ignore the domain part of the endpoints completely. For example, mystore.azconfig.io and mystore---abc.azconfig.azure.cn don't below to the same store.

Reply:
With the new naming convention for replicas. (i.e. `{storeName}-{replicaName}.azconfig.io`) checking whether all replicas belong to same primary configuration store has become difficult.
Consider we have the following list of replicas - 
1. my-store.azconfig.io
2. my-store-1.azconfig.io
3. my-store-1-1.azconfig.io

Now, all of the endpoints share the same substring `my-store` but we won't be able to discern whether they are primary stores or replicas.
1 and 3 could be different stores and 2 could be replica of 1.
OR
1 and 2 could be different stores and 3 could be replica of 2.

We would need to revisit how to validate this requirement.

Reply:
I _think_ the main reason we want to limit the endpoints to one store only is to prevent users from shooting their own feet. Data from different stores may not be in sync and the fail-over can cause unexpected results. Therefore, this validation will be just the best effort. Something like
- All endpoints must be different.
- All endpoints must be from the same domain.
- All endpoints must start with the same segment before the first dash.

Thoughts?

Reply:
I think we can have best effort validations here, but we would also need to document somewhere to the effect - "Endpoints to only one store are supported" or "Providing endoints to different stores may lead to undesirable results".

The combination of docs and best effort validation should prevent most of the mishaps.